libs/utils/WordPressUtils/src/test/java/org/wordpress/android/util/LogFileCleanerTest.kt

```diff
@@ -0,0 +1,70 @@
+package org.wordpress.android.util
+
+import android.content.Context
+import android.os.Build
+import androidx.test.core.app.ApplicationProvider
+import java.io.File
+import java.io.FileReader
+import kotlin.random.Random
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import org.wordpress.android.util.helpers.logfile.LogFileCleaner
+import org.wordpress.android.util.helpers.logfile.LogFileProvider
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.O_MR1])
+class LogFileCleanerTest {
+    lateinit var logFileProvider: LogFileProvider
+    private val maxFiles = 10
+
+    @Before
+    fun setup() {
+        val context: Context = ApplicationProvider.getApplicationContext()
+        logFileProvider = LogFileProvider.fromContext(context)
+
+        repeat(maxFiles) {
+            val file = File(logFileProvider.getLogFileDirectory(), "$it.log")
+            file.writeText("$it")
+            file.setLastModified((it * 10_000L).toLong())
+        }
+
+        assert(logFileProvider.getLogFileDirectory().listFiles().count() == maxFiles)
+    }
+
+    @After
+    fun tearDown() {
+        // Delete the test directory after each test
+        logFileProvider.getLogFileDirectory().deleteRecursively()
+    }
+
+    @Test
+    fun testThatCleanerPreservesMostRecentlyCreatedFiles() {
+        LogFileCleaner(logFileProvider, 3).clean()
+
+        val remainingFileIds = logFileProvider.getLogFileDirectory().listFiles().map {
+            FileReader(it).readText()
+        }.joinToString(",") // Strings are easier to assert against than arrays
```

Comment:
This can be written more concisely as following: (AS suggestion)

```suggestion
        val remainingFileIds = logFileProvider.getLogFileDirectory().listFiles().joinToString(",") {
            FileReader(it).readText()
        } // Strings are easier to assert against than arrays
```

Reply:
I did see this one, but I didn't really like it because I think it's less readable.

I'm not sure how to have AS stop suggesting it though, so I changed it in 3185b16a1144876ede635a895798f1b6663ed11e

😄

Reply:
Hmm.. interesting. I find it much more readable because this is a common type of operation in Kotlin 🤷‍♂